src/main/java/edu/harvard/iq/dataverse/api/SavedSearches.java

```diff
@@ -159,7 +197,7 @@ public Response add(JsonObject body) {
 
         try {
             SavedSearch persistedSavedSearch = savedSearchSvc.add(toPersist);
-            return ok("Added: " + persistedSavedSearch);
+            return ok("Added: " + persistedSavedSearch, Json.createObjectBuilder().add("id", persistedSavedSearch.getId()));
```

Comment:
If we're going to make a change here, should we go further and return a JSON object instead of a string? I do agree that something more easily parse-able would be great. These are the kinds of improvements that bring an API endpoint from experimental to more production ready.

Reply:
This is a JSON object and it contains the entity. 
```json 
{
    "status": "OK",
    "message": {
        "message": "Added: SavedSearch{id=4, query=*, savedSearchFilterQueries={[SavedSearchFilterQuery{id=2, filterQuery=subject_ss:Medicine, Health and Life Sciences}]}, definitionPoint=[Dataverse id:6 name:dv14b15aa1]}"
    },
    "data": {
        "id": 4
    }
}
```
but the structure is unparseable. 
I think it is required to have at least the id of the newly created entity in `data`. 
I would not know how to make a "good json object" in return as the whole API code has no consistency on the content `data`... But I agree with you it would be better to improve all json returns of APIs.